### PR TITLE
[IMP] website_forum: improve filters in forum

### PR DESCRIPTION
**Pull request by angv-odoo**
This commit improves filters in forum so that moderator can see all non-deleted posts on main page, use filters to move to flagged, closed or pending post lists, as well as see deleted posts and reopen them through filters if need arises.

task-3349373

### CODE REVIEW

File: addons/website_forum/views/forum_forum_templates_layout.xml [hunk 1]
```diff
@@ -46,11 +46,11 @@
                 <div class="d-flex gap-2 align-items-baseline">
                     <h3 t-attf-class="col-lg-10 mb-0" t-out="question.name"/>
                     <div class="col d-flex justify-content-end align-items-center">
-                        <i t-if="question.state == 'close'" class="fa fa-lock ms-2 fs-4" title="Closed" data-bs-toggle="tooltip" data-bs-placement="top"/>
-                        <span t-elif="not question.active" class="badge bg-danger">
-                            <t t-if="question.state!='offensive'">Deleted</t>
-                            <t t-if="question.state=='offensive'">Offensive</t>
-                            <t t-if="question.state=='offensive' and question.closed_reason_id">
+                        <i t-if="question.state == 'close'" class="fa fa-lock ms-2 fs-4" title="Closed" data-bs-toggle="tooltip" data-bs-placement="top"></i>
```
Comment (std-odoo) [inline addons/website_forum/views/forum_forum_templates_layout.xml]: why this change ?

File: addons/website_forum/models/forum_post.py [hunk 2]
```diff
@@ -324,7 +324,7 @@ def create(self, vals_list):
 
         for post in posts:
             # deleted or closed questions
-            if post.parent_id and (post.parent_id.state == 'close' or post.parent_id.active is False):
+            if post.parent_id and (post.parent_id.state in ['close', 'offensive'] or post.parent_id.active is False):
```
Comment (std-odoo) [inline addons/website_forum/models/forum_post.py]: picky std
```suggestion
            if post.parent_id and (post.parent_id.state in ('close', 'offensive') or post.parent_id.active is False):
```

File: addons/website_forum/controllers/website_forum.py [hunk 3]
```diff
@@ -107,12 +108,14 @@ def questions(self, forum, tag=None, page=1, filters='all', my=None, sorting=Non
 
         if not sorting:
             sorting = forum.default_order
+        values = self._prepare_user_values(forum=forum, searches=post, header={'ask_hide': not forum.active})
```
Comment (std-odoo) [inline addons/website_forum/controllers/website_forum.py]: what is `ask_hide` ?

**Review by std-odoo — COMMENTED** [review]
@angv-odoo Hi :) 

I have some minor comments, globally looks good

Also I couldn't test it because I get a 500 error when I open forum, can you rebase ?

Cheers

File: addons/website_forum/models/forum_post.py (hunk 2 above)
Comment (angv-odoo) [reply addons/website_forum/models/forum_post.py]: corrected, thanks!

File: addons/website_forum/views/forum_forum_templates_layout.xml (hunk 1 above)
Comment (angv-odoo) [reply addons/website_forum/views/forum_forum_templates_layout.xml]: not relevant anymore, so changing it back to how it was, thanks for pointing it out :)

File: addons/website_forum/controllers/website_forum.py (hunk 3 above)
Comment (angv-odoo) [reply addons/website_forum/controllers/website_forum.py]: something I don't need anymore, I got rid of this part :sweat_smile: 
Also: I corrected everything you pointed out, but during rebase I may have made an oopsie that may have caused closure of this draft pull request :sweat: could you check this draft pull request instead? https://github.com/odoo-dev/odoo/pull/2675  Sorry for the trouble and confusion :sweat:
